universal-login-relayer/test/integration/ethereum/WalletContractService.test.ts

```diff
@@ -36,6 +41,37 @@ describe('INT: WalletContractService', () => {
       refundReceiver: TEST_ACCOUNT_ADDRESS,
     };
     signedMessage = messageToSignedMessage(message, wallet.privateKey, 'istanbul', 'beta2');
+    fetchWalletVersionSpy = sinon.spy((walletContractService as any).blockchainService, 'fetchWalletVersion');
+  });
+
+  describe('version cache', () => {
+    it('throws an error if address is undefined', async () => {
+      await expect(walletContractService.getWalletVersion(undefined as any)).to.be.rejectedWith('Invalid address provided');
+    });
+
+    it('throws an error if address is empty', async () => {
+      await expect(walletContractService.getWalletVersion('')).to.be.rejectedWith('Invalid address provided');
+    });
+
+    it('returns proper wallet version and saves it to the cache', async () => {
```

Comment:
I would change the order of operations: 
- first, we call getWalletVersion, 
- then we check if it returns the proper version and stores it in walletVersion
- then we can call it again and check if it was called once

```ts
it('returns proper wallet version and saves it to the cache', async () => {
      const version = await walletContractService.getWalletVersion(proxyContract.address);
      expect(version).to.eq('beta2');
      expect((walletContractService as any).walletVersions).to.deep.eq({[proxyContract.address]: 'beta2'});
      await walletContractService.getWalletVersion(proxyContract.address);
      expect(fetchWalletVersionSpy).to.be.calledOnce;
    });
```